### PR TITLE
Fixed class.lescript_v2.php to work with ACMEv2

### PR DESCRIPTION
**Pull request by markuspetermann**
When requesting a certificate for multiple Domains/SANs, the request must be done in a single order.

### CODE REVIEW

File: install/updates/froxlor/0.9/update_0.9.inc.php [hunk 1]
```diff
@@ -3998,3 +3998,16 @@
 		updateToDbVersion('201805290');
 	}
 }
+
+if (isDatabaseVersion('201805290')) {
+
+	showUpdateStep("Adding leaccount field to panel customers");
+	Database::query("ALTER TABLE `froxlor`.`panel_customers` ADD COLUMN `leaccount` varchar(255) default '' AFTER `leregistered`;");
```
Comment (d00p) [inline install/updates/froxlor/0.9/update_0.9.inc.php]: do not used fixed database name please, just leave the `froxlor`.

**Comment by d00p** [comment]
Could you please explain briefly what the changes - escpecially the leaccount field - are doing and used for?

**Comment by markuspetermann** [comment]
The leaccount field stores the ACME account URL that is received once the account is registered. The ACME specification (https://tools.ietf.org/html/draft-ietf-acme-acme-14) states (p. 12):

"For newAccount requests, [...]. For all other requests, the request is signed using an existing account and there MUST be a "kid" field.  This field MUST contain the account URL received by POSTing to the newAccount resource."

Until now, the the account URL was only available when the account was registered, therefore all subsequent requests failed.

--

The second change fixes a bug when requesting a certificate for more than one subjectAltName (SAN), e.g. example.com / www.example.com. Until now a seperate order was issued for each SAN, which then failed at the end when requesting a certificate for all demains, while the order only contained the last domain.

The correct behaviour would be, to place one order per certificate and process authorization for all included SANs, as described in section 7.4 of the specification.

**Comment by d00p** [comment]
I assume this has been testen thoroughly? ;)

**Comment by markuspetermann** [comment]
Yes, this has been tested, no further issues so far.

**Pull request merged**